man/fi_cq.3.md

```diff
@@ -353,14 +353,23 @@ the the same as the count parameter.
 
 Returned source addressing data is converted from the native address
 used by the underlying fabric into an fi_addr_t, which may be used in
-transmit operations.  Returning fi_addr_t requires that the source
-address be inserted into the address vector associated with the
-receiving endpoint.  For applications using API version 1.5 and later
-(specified through the fi_getinfo call), if the source address has not
-been inserted into the address vector, fi_cq_readfrom will return
--FI_EAVAIL.  The completion will then be reported through
-fi_cq_readerr with error code -FI_EADDRNOTAVAIL.  See fi_cq_readerr
-for details.
+transmit operations.  Typically, returning fi_addr_t requires that
+the source address be inserted into the address vector associated with the
+receiving endpoint.  For endpoints allocated using the FI_SOURCE_ERR
+capability, if the source address has not been inserted into the
+address vector, fi_cq_readfrom will return -FI_EAVAIL.  The
+completion will then be reported through fi_cq_readerr with error
+code -FI_EADDRNOTAVAIL.  See fi_cq_readerr for details.
+
+If FI_SOURCE is specified without FI_SOURCE_ERR, source addresses
+which cannot be mapped to a local fi_addr_t will be reported as
+FI_ADDR_NOTAVAIL.  The behavior is dependent on the type of address
+vector in use.  For AVs of type FI_AV_MAP, source addresses may be
+mapped directly to an fi_addr_t value, even if the source address
+were not inserted into the AV.  This allows the provider to optimize
+the reporting of the source fi_addr_t without the overhead of
+verifying whether the address is in the AV.  If full address
+validation is necessary, FI_SOURCE_ERR must be used.
```

Comment:
So this `fi_addr_t` may not be inserted in any AV? Will this still get released when the AV is closed?